src/main/fc/init.c

```diff
@@ -521,22 +521,26 @@ void init(void)
     uartPinConfigure(serialPinConfig());
 #endif
 
+/*
+    TODO: validate this avoidance is still necessary
+    potential better solution would be to initialise and
+    configure the RX PPM / PWM devices before the uarts to
+    take ownership and avoid initialisation of the UARTs
+*/
 #if defined(AVOID_UART1_FOR_PWM_PPM)
 # define SERIALPORT_TO_AVOID SERIAL_PORT_USART1
 #elif defined(AVOID_UART2_FOR_PWM_PPM)
 # define SERIALPORT_TO_AVOID SERIAL_PORT_USART2
 #elif defined(AVOID_UART3_FOR_PWM_PPM)
 # define SERIALPORT_TO_AVOID SERIAL_PORT_USART3
 #endif
-    {
-        serialPortIdentifier_e serialPortToAvoid = SERIAL_PORT_NONE;
+    serialPortIdentifier_e serialPortToAvoid = SERIAL_PORT_NONE;
```

Comment:
this function is large. Using block makes scope of serialPortToAvoid smaller.

Reply:
Refactored away entirely.